jps-plugin/src/org/jetbrains/jps/incremental/scala/data/CompilerData.scala

```diff
@@ -38,8 +38,16 @@ object CompilerData {
 
     compilerJars.flatMap { jars =>
       val incrementalityType = SettingsManager.getProjectSettings(project.getProject).getIncrementalityType
-      javaHome(context, module).map(CompilerData(jars, _, incrementalityType))
+      if(SettingsManager.getHydraSettings(project.getProject).isHydraEnabled && SettingsManager.getHydraSettings(project.getProject).getArtifactPaths.containsKey(version(jars.get.compiler).getOrElse("UNKNOWN"))) {
+        val scalaVersion = version(jars.get.compiler)
+        val hydraData = HydraData(project.getProject)
+        val hydraJars = Some(new CompilerJars(jars.get.library, hydraData.getCompilerJar(scalaVersion.get).getOrElse(jars.get.compiler), hydraData.otherFiles(scalaVersion.get)))
+        javaHome(context, module).map(CompilerData(hydraJars, _, incrementalityType))
+      } else {
+        javaHome(context, module).map(CompilerData(jars, _, incrementalityType))
```

Comment:
The call to `javaHome` is common to both branches and could be extracted out. You could have:
```
val compilerJars = {
  if(SettingsManager.getHydraSettings(project.getProject).isHydraEnabled && SettingsManager.getHydraSettings(project.getProject).getArtifactPaths.containsKey(version(jars.get.compiler).getOrElse("UNKNOWN"))) {
    // hydra specific code
  }
  else jars
}
javaHome(context, module).map(CompilerData(compilerJars, _, incrementalityType))
```